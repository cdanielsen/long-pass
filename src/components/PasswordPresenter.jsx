import React from 'react'
import PropTypes from 'prop-types'

function PasswordPresenter({ id, passwordContent }) {
  const styledWord = passwordContent.join(' ')
  return <div id={id}>{styledWord}</div>
}

PasswordPresenter.propTypes = {
  passwordContent: PropTypes.array,
  id: PropTypes.string,
}

export default PasswordPresenter
