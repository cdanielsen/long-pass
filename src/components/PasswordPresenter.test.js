import React from 'react'
import PasswordPresenter from './PasswordPresenter.jsx'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'

configure({adapter: new Adapter()})

describe('PasswordPresenter component', () => {
  const props = {
    id: 'presenter',
    passwordContent: ['force', 'hammer', 'flower', 'coffee']
  }
  test('renders a div with the passed in password content', () => {
    const wrapper = shallow(<PasswordPresenter {...props} />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.prop('id')).toEqual(props.id)
    expect(wrapper.text()).toEqual(props.passwordContent.join(' '))
  })
})
