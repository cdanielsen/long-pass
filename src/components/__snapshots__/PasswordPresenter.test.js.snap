// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordPresenter component renders a div with the passed in password content 1`] = `
<div
  id="presenter"
>
  force hammer flower coffee
</div>
`;
